atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -77,6 +80,24 @@ public void registerMetric(Class clazz, String metricName, Gauge gauge, Map<Stri
                 gauge);
     }
 
+    public void registerGaugeForTable(Class clazz, String metricName, TableReference tableRef, Gauge gauge) {
+        Map<String, String> tag = getTagFor(tableRef);
+
+        taggedMetricRegistry.gauge(MetricName.builder()
+                        .safeName(MetricRegistry.name(clazz, metricName))
+                        .safeTags(tag)
+                        .build(),
+                gauge);
+    }
+
+    private Map<String, String> getTagFor(TableReference tableRef) {
```

Comment:
This should probably live in LoggingArgs. Also, I'd use `LoggingArgs.safeTableOrPlaceholder` to avoid custom conditional logic below

Reply:
`LoggingArgs.safeTableOrPlaceholder` is much nicer, thanks.

Going to leave here so the LoggingArgs logic doesn't know anything about metrics and tags.